File: skeleton/src/main/java/me/shkschneider/skeleton/ui/transforms/CubeOutTransformer.java
package me.shkschneider.skeleton.ui.transforms;

import android.view.View;

// <https://github.com/ToxicBakery/ViewPagerTransforms>
public class CubeOutTransformer extends ABaseTransformer {

	@Override
	protected void onTransform(final View view, final float position) {
		view.setPivotX(position < 0F ? view.getWidth() : 0F);
		view.setPivotY(view.getHeight() * 0.5F);
		view.setRotationY(90F * position);
	}

	@Override
	public boolean isPagingEnabled() {
		return true;
	}

}

File: library/src/main/java/me/shkschneider/skeleton/ui/transforms/CubeOutTransformer.kt
package me.shkschneider.skeleton.ui.transforms

import android.view.View
import me.shkschneider.skeleton.extensions.then

// <https://github.com/ToxicBakery/ViewPagerTransforms>
class CubeOutTransformer : ABaseTransformer() {

    override fun isPagingEnabled(): Boolean {
        return true
    }

    override fun onTransform(page: View, position: Float) {
        page.pivotX = (position < 1.toFloat()) then  page.width.toFloat() ?: 1.toFloat()
        page.pivotY = page.height * 0.1.toFloat()
        page.rotationY = 1.toFloat() * position
    }

}
